owncloudApp/src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java

```diff
@@ -193,9 +190,10 @@ public void sendDownloadedFile(OCFile file) {
             sendIntent.putExtra(Intent.ACTION_SEND, true);      // Send Action
 
             // Show dialog, without the own app
-            String[] packagesToExclude = new String[]{mFileActivity.getPackageName()};
-            DialogFragment chooserDialog = ShareLinkToDialog.newInstance(sendIntent, packagesToExclude);
-            chooserDialog.show(mFileActivity.getSupportFragmentManager(), FTAG_CHOOSER_DIALOG);
+            ShareLinkToShareSheet shareLinkToShareSheet = ShareLinkToShareSheet.Companion.getInstance(sendIntent, mFileActivity, componentName -> {
+                return componentName.getPackageName().equals(mFileActivity.getPackageName());
+            });
+            shareLinkToShareSheet.show();
```

Comment:
Similar to sharing text, you can simply use
`            mFileActivity.startActivity(Intent.createChooser(sendIntent, null));
`
And the Sharesheet will open without problems.

Taking into account that currently we can not share multiple files, you don't need to take care of [that](https://developer.android.com/training/sharing/send#share-multiple-content).

Adding [rich content to text previews](https://developer.android.com/training/sharing/send#adding-rich-content-previews) would be interesting, but not needed for now.

And similar to text, we need to take care of:
- Do we need to add ownCloud as target? I don't think so, but maybe i'm missing a special usecase..

If you want to exclude specific targets, you can use [EXTRA_EXCLUDE_COMPONENTS](https://developer.android.com/training/sharing/send#excluding-specific-targets-by-component)

Reply:
Hello @abelgardep  
i have tried to apply this approch at my first try but i had to twist it abit for the following reasons :
 1-- Intent.EXTRA_EXCLUDE_COMPONENTS not  support android below < 24
       so we had to filter Target intents and add the desired list ، that is why we have to **Explicitly** add CopyToClipBoard 
      intent 
2--honstly i wasn't fully aware of sharing  (single/multiple) files Logic as mentioned 👀   ,  as we filtering target intent 
      ,i was just following the same logic the old dialog was build on

appreciate your review , and waiting your feedback.....

Reply:
1- It would be great to use EXTRA_EXCLUDE_COMPONENTS for api>=24 and another behaviour for api<24. That way if we want to upgrade sdk in the future, we could simply remove old code instead of replace it again.

2- Yes, don't worry about that, it is enough to keep the previous functionality

Yesterday i tried your solution and i think that there were not as many apps as there were before. With the code i posted in my previous comment, i was able to share with some apps that i wasn't able with this solution.

Taking into account that it may add complexity to this task, i think it would be enough to use sharesheet with api>=24 using EXTRA_EXCLUDE_COMPONENTS and keep using old ShareLinkToDialog when api <24.

What do you think? Thanks for your interest 🥇

Reply:
i think your soultion is producing more target Intent cause we are not specifing the **MiMe** type 
according to the **old ShareDialog** , it was required to query intents with flag  MATCH_DEFAULT_ONLY to get most supported TargetIntetn 
` val resolveInfoList =
            fileActivity?.baseContext?.packageManager?.queryIntentActivities(intentToShareLink, PackageManager.MATCH_DEFAULT_ONLY)`

--- and having both shareDialog and shareSheet would't add complexty more than code duplacity 
      and one of the main purpose this ticket opened for was to remove dialog completly even from the code base and 
       only use shareSheet.

Reply:
I know that one of the main purposes of this ticket was to remove ShareLinkToDialog. But taking into account that we cannot keep the same behaviour for <24 devices, i would not remove it yet. I think that the best solution for the moment is to check SDK and use old implementation for <24 and sharesheet for >=24. Something like:
```
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
        useShareSheet()
    } else {
        useShareLinkToDialog()
    }
```

Reply:
Hi @ddopik in Android 7.0, there are no big differences between this two implementations.  But there are big differences in Android 10, where i can only share with 2 apps with this implementation which is not the expected behavior.

Using Android 7.0: 
| This solution | Using the one i suggested | 
| ------------- | ------------- |
![current_7](https://user-images.githubusercontent.com/47524927/89514221-df257000-d7d5-11ea-99e2-771a69506e3c.png) | ![android-text](https://user-images.githubusercontent.com/47524927/84505129-7cc05f80-acbd-11ea-8355-2f4c63d2abcf.jpg)|

Reply:
i have investigated in your issue when you said that you have only 2 app to share to when runnning on android(10) Q , and you are totaly right , i spent some time depugging and testign code again but meanwhile i have found this issue openedin [stackOverFlow](https://stackoverflow.com/questions/59786777/why-intent-extra-initial-intents-not-working-in-android-10) and also in [IssueTracker](https://issuetracker.google.com/issues/134367295). 

i think we shoud implement our own bottomSheet to match the desired behaviour and gurantee the compatibility 
what do you think @abelgardep @jesmrec @michaelstingl

Reply:
I would go with a solution like the one i suggested you some days ago: https://github.com/owncloud/android/pull/2922#discussion_r463425724 

```   
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
    useShareSheet()
} else {
    useShareLinkToDialog()
}
```

Using native sharesheet with EXTRA_EXCLUDE_COMPONENTS for ownCloud when api>24 and the current implementation when api<24. Don't you think?

Reply:
Moving stuff to native is something where we are moving to (SAF is the best example). If we can use the native implementation, i think it'd be better.

Reply:
@ddopik what do you think about @abelgardep suggestion? we can take advantage of the native implementation from android N on, and let the old one for users with older devices. Once we upgrade our minimum API to 24(somewhen in the future), we can get rid of the old implementation and keep the feature as native. We can help you, whatever you need.